core/consensus/babe/impl/babe_lottery_impl.hpp

```diff
@@ -21,42 +21,44 @@ namespace kagome::consensus::babe {
 namespace kagome::crypto {
   class Hasher;
   class VRFProvider;
+  class SessionKeys;
 }  // namespace kagome::crypto
 
 namespace kagome::consensus::babe {
 
   class BabeLotteryImpl : public BabeLottery {
    public:
-    BabeLotteryImpl(std::shared_ptr<crypto::VRFProvider> vrf_provider,
-                    std::shared_ptr<BabeConfigRepository> babe_config_repo,
+    BabeLotteryImpl(std::shared_ptr<BabeConfigRepository> config_repo,
+                    std::shared_ptr<crypto::SessionKeys> session_keys,
+                    std::shared_ptr<crypto::VRFProvider> vrf_provider,
                     std::shared_ptr<crypto::Hasher> hasher);
 
-    void changeEpoch(EpochNumber epoch,
-                     const Randomness &randomness,
-                     const Threshold &threshold,
-                     const crypto::Sr25519Keypair &keypair) override;
-
     EpochNumber getEpoch() const override;
 
-    std::optional<crypto::VRFOutput> getSlotLeadership(
-        SlotNumber i) const override;
-
-    crypto::VRFOutput slotVrfSignature(SlotNumber slot) const override;
+    bool changeEpoch(EpochNumber epoch,
```

Comment:
Add docs with explanation of params and returned value please

Reply:
Already exists in base class: BabeLottery